Base/src/test/java/io/deephaven/base/FileUtilsTest.java

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.base;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+
+import junit.framework.TestCase;
+import org.junit.Assert;
+
+public class FileUtilsTest extends TestCase {
+
+    public void testConvertToFileURI() throws IOException {
+        final File currentDir = new File("").getAbsoluteFile();
+        fileUriTestHelper(currentDir.toString(), true, currentDir.toURI().toString());
+
+        final File someFile = new File(currentDir, "tempFile.txt");
+        fileUriTestHelper(someFile.getPath(), false, someFile.toURI().toString());
+
+        // Check if trailing slash gets added for a directory
+        final String expectedURI = "file:" + currentDir.getPath() + "/path/to/directory/";
+        fileUriTestHelper(currentDir.getPath() + "/path/to/directory", true, expectedURI);
+
+        // Check if multiple slashes get normalized
+        fileUriTestHelper(currentDir.getPath() + "////path///to////directory", true, expectedURI);
+
+        // Check if multiple slashes in the beginning get normalized
+        fileUriTestHelper("////" + currentDir.getPath() + "/path/to/directory", true, expectedURI);
+    }
+
+    private static void fileUriTestHelper(final String filePath, final boolean isDirectory,
+            final String expectedURIString) {
+        Assert.assertEquals(expectedURIString, FileUtils.convertToURI(filePath, isDirectory).toString());
+        Assert.assertEquals(expectedURIString, FileUtils.convertToURI(new File(filePath), isDirectory).toString());
+        Assert.assertEquals(expectedURIString, FileUtils.convertToURI(Path.of(filePath), isDirectory).toString());
+    }
+
+    public void testConvertToS3URI() throws URISyntaxException {
+        assertEquals("s3://bucket/key", FileUtils.convertToURI("s3://bucket/key", false).toString());
```

Comment:
We should test `convertToURI("s3://bucket/key/", false)` and `convertToURI("s3://bucket/key//", false)`. I don't know what we want / expect in this situation. Similar Q for files.

Reply:
I think these would be bad inputs. I can maybe raise an exception in this case.